### ISSUE TITLE
fparser2 produces no output if program and end program names do not match

### ISSUE THREAD

**Issue by rupertford** [stfc/fparser#125]

**Comment by rupertford** [comment]
```
rupert@ubuntu:~/proj/fparser/src/fparser$ cat test.f90
program aa
end program bb
rupert@ubuntu:~/proj/fparser/src/fparser$ ./scripts/fparser2.py test.f90
rupert@ubuntu:~/proj/fparser/src/fparser$ 
```

**Comment by rupertford** [comment]
The error is due to code at line 514 in utils.py on master. If this is commented out then the code is output but no error is raised. I need to replace this code with a FortranSyntaxError. However, PR #121 has modified this region of code so I'll hold off making the change until that is on master.

**Comment by rupertford** [comment]
Looking at PR #121, it may be preferable for Main_Program match() to set match_names to True and have the logic for that (once PR #121 is on master) to deal with the check. If we do this then I think it implies that the code that checks for matches further down (that I was suggesting to modify in the previous comment) could be removed completely.

**Comment by rupertford** [comment]
Created branch prognames_nomatch_error

**Comment by TeranIvy** [comment]
Merged #137 so closing this issue.
